Data Security: update Agent Trace obfuscation (#19656)
DataDog/documentation: content/en/tracing/configure_data_security/_index.md

### Resource names

Datadog spans include a resource name attribute that may contain sensitive data. The Datadog Agent implements obfuscation of resource names for several known cases:

* **SQL numeric literals and bind variables are obfuscated**: For example, the following query `SELECT data FROM table WHERE key=123 LIMIT 10` is obfuscated to `SELECT data FROM table WHERE key = ? LIMIT ?` before setting the resource name for the query span.
* **SQL literal strings are identified using standard ANSI SQL quotes**: This means strings should be surrounded in single quotes (`'`). Some SQL variants optionally support double-quotes (`"`) for strings, but most treat double-quoted things as identifiers. The Datadog obfuscator treats these as identifiers rather than strings and does not obfuscate them.
* **Redis queries are quantized by selecting only command tokens**: For example, the following query `MULTI\nSET k1 v1\nSET k2 v2` is quantized to `MULTI SET SET`.

### Trace obfuscation

> The Agent can obfuscate other sensitive data in [traces][2] that is not within the resource name. Configuration of obfuscation rules can be done with environment variables or the `datadog.yaml` configuration file.

Reviewer: ```suggestion
The Agent can obfuscate other sensitive data in [traces][2] that is not within the resource name. You can configure the obfuscation rules using environment variables or the `datadog.yaml` configuration file.
```

  ## (...)

  obfuscation:
    mongodb:
      ## Enables obfuscation rules for spans of type "mongodb". Enabled by default.
      enabled: true
      keep_values:
        - document_id
        - template_id
      obfuscate_sql_values:
        - val1
```

This can also be disabled with the environment variable `DD_APM_OBFUSCATION_MONGODB_ENABLED=false`.

> * `keep_values` or environment variable `DD_APM_OBFUSCATION_MONGODB_KEEP_VALUES` - defines a set of keys to exclude from Agent trace obfuscation. If not set, all keys will be obfuscated.

Reviewer: ```suggestion
* `keep_values` or environment variable `DD_APM_OBFUSCATION_MONGODB_KEEP_VALUES` - defines a set of keys to exclude from Agent trace obfuscation. If not set, all keys are obfuscated.
```

> * `obfuscate_sql_values` or environment variable `DD_APM_OBFUSCATION_MONGODB_OBFUSCATE_SQL_VALUES` - defines a set of keys to include Agent trace obfuscation. If not set, all keys will be obfuscated.

Reviewer: ```suggestion
* `obfuscate_sql_values` or environment variable `DD_APM_OBFUSCATION_MONGODB_OBFUSCATE_SQL_VALUES` - defines a set of keys to include Agent trace obfuscation. If not set, all keys are obfuscated.
```

  ## (...)

  obfuscation:
    elasticsearch:
      ## Enables obfuscation rules for spans of type "elasticsearch". Enabled by default.
      enabled: true
      keep_values:
        - client_id
        - product_id
      obfuscate_sql_values:
        - val1
```

This can also be disabled with the environment variable `DD_APM_OBFUSCATION_ELASTICSEARCH_ENABLED=false`.

> * `keep_values` or environment variable `DD_APM_OBFUSCATION_ELASTICSEARCH_KEEP_VALUES` - defines a set of keys to exclude from Agent trace obfuscation. If not set, all keys will be obfuscated.

Reviewer: ```suggestion
* `keep_values` or environment variable `DD_APM_OBFUSCATION_ELASTICSEARCH_KEEP_VALUES` - defines a set of keys to exclude from Agent trace obfuscation. If not set, all keys are obfuscated.
```

> * `obfuscate_sql_values` or environment variable `DD_APM_OBFUSCATION_ELASTICSEARCH_OBFUSCATE_SQL_VALUES` - defines a set of keys to include Agent trace obfuscation. If not set, all keys will be obfuscated.

Reviewer: ```suggestion
* `obfuscate_sql_values` or environment variable `DD_APM_OBFUSCATION_ELASTICSEARCH_OBFUSCATE_SQL_VALUES` - defines a set of keys to include Agent trace obfuscation. If not set, all keys are obfuscated.
```
